tests/src/test/scala/cats/tests/ValidatedTests.scala

```diff
@@ -188,4 +188,20 @@ class ValidatedTests extends CatsSuite {
     val z = x.map2(y)((i, b) => if (b) i + 1 else i)
     z should === (NonEmptyList("error 1", "error 2").invalid[Int])
   }
+
+  test("ensure on Invalid is identity") {
+    forAll { (x: Validated[Int,String], i: Int, p: String => Boolean) =>
+      if (x.isInvalid) {
+        x.ensure(i)(p) should === (x)
+      }
+    }
+  }
+
+  test("ensure should fail if predicate not satisfied") {
+    forAll { (x: Validated[String, Int], s: String, p: Int => Boolean) =>
+      if (x.isValid && !p(x getOrElse 0)) {
```

Comment:
Alternatively I think this could be `if (x.exists(i => !(p(i))))`. What's here is totally fine for a test though.